### DISCUSSION TITLE
update schedule_interval param through Airflow api failed

### DISCUSSION THREAD

**Discussion by davidbelhamou** [apache/airflow#26522] [Q&A] [answered]
Hello,

I', trying to update the schedule interval parameter for one of my dag through Airflow api, but i'm getting an error
here see the http response i get : 
{
    "detail": "Property is read-only - 'schedule_interval'",
    "status": 400,
    "title": "Bad Request",
    "type": "https://airflow.apache.org/docs/apache-airflow/2.3.4/stable-rest-api-ref.html#section/Errors/BadRequest"
}

i made a patch request via postman:
url : http://my-url/api/v1/dags/my_dag
body: (json) -> {"schedule_interval": "50 16 * * MON"}

when updating the is_paused field its work fine but how can i make it work for the schedule_interval field.

Thanks

**Comment by potiuk** [comment, marked as answer]
You can't. Schedule interval is "read-only" as you see. It can only be changed by modifying the DAG Python source code - this is how Airlfow works.

**Reply by davidbelhamou** [reply]
Thanks

**Reply by moxgeek** [reply]
ok, but why ? what's the issue if i wanna change my schedule using the API ? why it's not allowed ?

**Reply by potiuk** [reply]
Because it's part of the code definition. Maybe in the future we will decouple the schedule from DAG definition. It's been discussed - but they are coupled now. If you would like to propose changing it - feel free to propose  Airflow Improvement Proposal (see community tab in Airflow website) - and propose changing it for Airlfow 3 (because Airflow 2 is not going to get any new features - only Airlfow 3 - which we work on - will get new changes).

And yes I think several people raised that as a possible feature, so it is indeed worth discussing and proposing changing it - so if you feel like, it you are welcome to start such discussion.
